db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/FromBaseTable.java

```diff
@@ -1813,13 +1896,32 @@ else if(isOneRowResultSet()){
          * Splice changed the below loop to iterate over storeRestrictionList instead of restrictionList */
         for(int i=0;i<storeRestrictionList.size();i++){
             Predicate pred=storeRestrictionList.elementAt(i);
-            if(pred.isInListProbePredicate() && pred.isStartKey()){
-                disableBulkFetch();
-                multiProbing=true;
-                break;
+            if(pred.isStartKey()){
+                if (pred.isInListProbePredicate()) {
+                    disableBulkFetch();
+                    multiProbing = true;
+                }
+            }
+            if (pred.isStartKey() || pred.isStopKey()){
+                if (numUnusedLeadingIndexFields == 0) {
+                    numUnusedLeadingIndexFields = ap.getNumUnusedLeadingIndexFields();
+                    if (numUnusedLeadingIndexFields >= 1) {
+                        currentIndexFirstColumnStats = trulyTheBestConglomerateDescriptor.getFirstColumnStats();
+                        if (numUnusedLeadingIndexFields > 1)
+                            throw StandardException.newException(LANG_INTERNAL_ERROR,
+                               "IndexPrefixIteratorMode currently allows at most one leading index column to be unspecified in predicates.");
```

Comment:
This is the first question I have in mind after reading @msirek 's comment to Jira. Does it make sense to extend the support to multiple leading columns? I guess it's a question about theory vs. reality. Probably it's not common that the number of distinct values in all leading columns are small?

Reply:
Yes, we'll probably want to wait and see if there is a customer need for it.  It possibly could occur if the first column has a very small number of values, like a region code {west, south, east, north}, or a 2-value field like history rows vs. current rows.

Reply:
OK. That makes sense.